Modules/TRD/src/PulseHeightTrackMatch.cxx

```diff
@@ -0,0 +1,308 @@
+                [&digitv](unsigned int i, unsigned int j) { return digitIndexCompare_phtm(i, j, digitv); });
+
+      for (const auto& triggerTracks : triggerrecordstracks) {
+
+        if (trigger.getBCData() != triggerTracks.getBCData()) {
+          continue;
+        }
+
+        int nTracks = 0;
+        for (int iTrack = triggerTracks.getFirstTrack(); iTrack < triggerTracks.getFirstTrack() + triggerTracks.getNumberOfTracks(); iTrack++) {
+          nTracks++;
+          const auto& track = tracks[iTrack];
+
+          // printf("Got track with %i tracklets \n", track.getNtracklets());
+          int ntracklets = track.getNtracklets();
+          if (ntracklets > 10)
+            mTrackletsPerMatchedTrack->Fill(9);
+          else
+            mTrackletsPerMatchedTrack->Fill(ntracklets);
+
+          for (int iLayer = 0; iLayer < 6; iLayer++) {
+            int trackletIndex = track.getTrackletIndex(iLayer);
+
+            if (trackletIndex == -1) {
+              continue;
+            }
+
+            const auto& trklt = tracklets[trackletIndex];
+
+            int det = trklt.getDetector();
+            int sec = det / 30;
+            int row = trklt.getPadRow();
+            // int col = trklt.getPadCol();
+
+            // obtain pad number relative to MCM center
+            int padLocal = trklt.getPositionBinSigned() * o2::trd::constants::GRANULARITYTRKLPOS;
+            // MCM number in column direction (0..7)
+            int mcmCol = (trklt.getMCM() % o2::trd::constants::NMCMROBINCOL) + o2::trd::constants::NMCMROBINCOL * (trklt.getROB() % 2);
+            int col = 6.f + mcmCol * ((float)o2::trd::constants::NCOLMCM) + padLocal;
+
+            for (int iDigit = trigger.getFirstDigit() + 1; iDigit < trigger.getFirstDigit() + trigger.getNumberOfDigits() - 1; ++iDigit)
+
+            {
+              const auto& digit = digits[iDigit];
+              const auto& digitBefore = digits[iDigit - 1];
+              const auto& digitAfter = digits[iDigit + 1];
+
+              std::tuple<unsigned int, unsigned int, unsigned int> aa, ba, ca;
+              aa = std::make_tuple(digitBefore.getDetector(), digitBefore.getPadRow(), digitBefore.getPadCol());
+              ba = std::make_tuple(digit.getDetector(), digit.getPadRow(), digit.getPadCol());
+              ca = std::make_tuple(digitAfter.getDetector(), digitAfter.getPadRow(), digitAfter.getPadCol());
+
+              auto [det1, row1, col1] = aa;
+              auto [det2, row2, col2] = ba;
+              auto [det3, row3, col3] = ca;
+
+              bool consecutive = false;
+              bool digitmatch = false;
+
+              if (det == det2 && row == row2) {
+
+                int coldif = col2 - col;
+                if (TMath::Abs(coldif) < 4)
+                  digitmatch = true;
+              }
+              if (det1 == det2 && det2 == det3 && row1 == row2 && row2 == row3 && col2 + 1 == col1 && col3 + 1 == col2)
```

Comment:
There is some redundancy here, because the digits are sorted. For example, det1==det3 automatically implies det1==det2 and det2==det3.